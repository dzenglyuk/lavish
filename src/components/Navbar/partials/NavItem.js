import React from "react";
import { Link } from "react-router-dom";

const NavItem = ({ title, path, handleClick }) => {
    return (
        <li className="nav-item" onClick={handleClick}>
            <Link to={path} className="nav-links" data-testid="nav-item-link">
                {title}
            </Link>
        </li>
    );
};

NavItem.defaultProps = {
    title: "Home",
    path: "/",
};

export default React.memo(NavItem);
