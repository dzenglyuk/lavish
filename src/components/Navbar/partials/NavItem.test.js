import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { StaticRouter as Router } from "react-router";

import NavItem from "./NavItem";

test("Renders correctly", () => {
    const { getByText, getByTestId } = render(
        <Router>
            <NavItem />
        </Router>
    );

    getByText("Home");
    expect(getByTestId("nav-item-link")).toHaveProperty(
        "href",
        "http://localhost/"
    );
});

test("Render correctly with props", () => {
    const { getByText, getByTestId } = render(
        <Router>
            <NavItem title={"Test Title"} path={"/somepath"} />
        </Router>
    );

    getByText("Test Title");
    expect(getByTestId("nav-item-link")).toHaveProperty(
        "href",
        "http://localhost/somepath"
    );
});

test("Matches the snapshot 1", () => {
    const tree = renderer
        .create(
            <Router>
                <NavItem />
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test("Matches the snapshot 2", () => {
    const tree = renderer
        .create(
            <Router>
                <NavItem title="Services" path="/services" />
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
