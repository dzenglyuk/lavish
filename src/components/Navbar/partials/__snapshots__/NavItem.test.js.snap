// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-links"
    data-testid="nav-item-link"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</li>
`;

exports[`Matches the snapshot 2 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-links"
    data-testid="nav-item-link"
    href="/services"
    onClick={[Function]}
  >
    Services
  </a>
</li>
`;
